pycurl_requests/tests/utils.py

```diff
@@ -93,6 +97,44 @@ def do_GET_response_headers(self):
     def do_HTTP_404(self):
         self.send_error(404, 'Not Found')
 
+    def do_POST(self):
+        path = self.url.path[1:].replace('/', '_')
+        getattr(self, f'do_POST_{path}', self.do_HTTP_404)()
+
+    def do_POST_stream(self):
+        self.POST_stream_helper(allow_chunked=True)
+
+    def do_POST_stream_no_chunked(self):
+        self.POST_stream_helper(allow_chunked=False)
+
+    def POST_stream_helper(self, allow_chunked: bool):
+        if "Content-Length" in self.headers:
+            content_length = int(self.headers["Content-Length"])
+            body = self.rfile.read(content_length)
+        elif "Transfer-Encoding" in self.headers and "chunked" in self.headers["Transfer-Encoding"]:
+            if not allow_chunked:
+                self.response('This endpoint has chunked transfer deactivated.', status=(400, "Bad Request"))
+                return
+            body = b""
```

Comment:
Joining immutable `bytes` objects can be potentially expensive. [`bytearray`](https://docs.python.org/3/library/stdtypes.html#bytearray) is a better choice if you need something mutable.